packages/@tinacms/graphql/src/database/index.ts

```diff
@@ -109,12 +103,71 @@ export class Database {
     }
   }
 
+  public addPendingDocument = async (
+    filepath: string,
+    data: { [key: string]: unknown }
+  ) => {
+    const { stringifiedFile, payload } = await this.stringifyFile(
+      filepath,
+      data
+    )
+    if (this.store.supportsSeeding()) {
+      await this.bridge.put(filepath, stringifiedFile)
+    }
+    await this.store.put(filepath, payload)
+  }
+
   public put = async (filepath: string, data: { [key: string]: unknown }) => {
     if (SYSTEM_FILES.includes(filepath)) {
-      await this.bridge.put(
-        path.join(GENERATED_FOLDER, `${filepath}.json`),
-        JSON.stringify(data, null, 2)
+      throw new Error(`Unexpected put for config file ${filepath}`)
+    } else {
+      const { stringifiedFile, payload } = await this.stringifyFile(
+        filepath,
+        data
       )
+      if (this.store.supportsSeeding()) {
+        await this.bridge.put(filepath, stringifiedFile)
+      }
+      const existingData = await this.get<{ _collection: string }>(filepath)
+      const collection = this.tinaSchema.getCollection(existingData._collection)
+      const atts = await _indexCollectable({
+        record: filepath,
+        value: existingData,
+        field: collection,
+        prefix: collection.name,
+        database: this,
+      })
+      const atts2 = await _indexCollectable({
+        record: filepath,
+        value: data,
+        field: collection,
+        prefix: collection.name,
+        database: this,
+      })
+      await sequential(atts, async (attribute) => {
+        const records = (await this.store.get<string[]>(attribute)) || []
+        await this.store.put(
+          attribute,
+          records.filter((item) => item !== filepath)
+        )
+        return true
+      })
+      await sequential(atts2, async (attribute) => {
+        const records = (await this.store.get<string[]>(attribute)) || []
+        await this.store.put(attribute, [...records, filepath])
+        return true
+      })
```

Comment:
This goes through and removes the 'metadata' attributes, and re-adds the new ones.